### DISCUSSION TITLE
Occasionally getting Error from upload stream

### DISCUSSION THREAD

**Discussion by ericoporto** [cirruslabs/cirrus-ci-docs#1075] [Q&A] [answered]
Hi,

We are occasionally getting the "Error from upload stream", noted here: https://github.com/adventuregamestudio/ags/issues/1801

This specific one is here: https://api.cirrus-ci.com/v1/task/4869075909214208/logs/proj.log

but it's different from the previous reported issue here (artifacts are clearly found): https://github.com/cirruslabs/cirrus-ci-docs/issues/913

Any recommendations? Is there a way to force the artifact upload task to fail if no artifact is uploaded? In this way we can still rerun the task.

The problem isn't the failure, the problem is that the upload task doesn't produce an error code on failure and thus it's considered successful (green), making hard to understand why something is not right later.

This maybe should be a bug to be reported/solved in [cirrus-cli](https://github.com/cirruslabs/cirrus-cli).

Thanks!

**Comment by ericoporto** [comment, marked as answer]
Just to clarify, the problem is not the error, the problem is the task succeeded, WITH the error!

**Reply by fkorotkov** [reply]
Interesting! But seems both artifacts are uploaded successfully... There were some retries in place which might've print an error but retry. We've also just revamped the artifact uploading to make is faster and more resilient to network issues in https://github.com/cirruslabs/cirrus-ci-agent/pull/263. We'll rollout the new agent later this week.

**Reply by ericoporto** [reply]
Thanks, I will leave at this and report if it happens again! I have a new question, but I will open a new entry here. :)
